Bloom.md

```diff
@@ -0,0 +1,341 @@
+
+
+## Design Considerations
+
+The ValkeyBloom module brings in a bloom module data type into Valkey and provides commands to create / reserve
+bloom filters, operate on them (add items, check if items exist), inspect bloom filters, etc.
+It allows customization of properties of bloom filter (capacity, false positive rate, expansion rate, specification of
+scaling vs non-scaling filters etc) through commands and configurations. It also allows users to create scalable bloom
+filters and back up & restore bloom filters (through RDB load and save).
+
+ValkeyBloom provides commands (BF.*), configs, etc to operate on Bloom objects which are top-level structures containing
+lower-level BloomFilter provided by an external crate. ValkeyBloom utilizes an open source (BSD-2) Bloom Filter Rust library
+around which it implements a scalable bloom filter.
+
+When a bloom filter is created, a bit array is created with a length proportional to the capacity (number of items the user
+wants to add to the filter) and hash functions are also created. The number of hash functions are controlled by the
+false positive rate that the user configures.
+
+When a user adds an item (e.g. BF.ADD) to the filter, the item is passed through the filter and corresponding bit are set
+to 1. When a user checks whether an item exists on a filter (e.g. BF.EXISTS), the item is passed through the filters
+and if all the resolved bits have values as 1, we can say that the item exists with a false positive rate of
+X (specified by the user when creating the filter). If any of the bits are 0, the item does not exist and the BF.EXISTS
+operation will return 0.
+
+We have the following terminologies / properties:
+* Bloom Object: The top level structure representing the data type. It contains meta data and a list of lower-level
+                Bloom Filters (Implemented by an external crate) in case of scaling and a single lower-level bloom filter
+                in case of non scaling.
+* Bloom Filter: A single bloom filter (Implemented by an external crate).
+* Capacity: The number of items we expect to add to a bloom filter. This controls the size of the filter.
+* False Positive Rate: The accuracy the user expects when operating (set / check) on a filter. This controls the number
+                        of hash functions.
+* Expansion Rate: This is used in scalable bloom filters where multiple bloom filters are stacked to allow users to
+                    continue using the same bloom object when it reaches capacity by adding another filter of larger
+                    capacity (expansion rate * prev filter capacity).
+* Hash Functions: Hash functions used for bit check and set operations underneath.
+* Hash Key: Keys used by the hashing function.
+
+
+### Module OnLoad
+
+Upon loading, the module registers a new bloom module based data type, creates bloom filter (BF.*) commands,
+bloom specific configurations and the bloom ACL category.
+
+* Module name: bloom
```

Comment:
Can this conflict with other oss bloom module ? would prefer `valkeybloom`

Reply:
i see in json we are doing this:
```
Upon loading, the module registers a new JSON data type. All operations such as query, insert, update and delete are 
efficiently performed on the in-memory document objects, as opposed to JSON text.
* Module name: json
* JSON data type name: ReJSON-RL (Note: We use the same name as the one in RedisJSON for the sake of RDB compatibility.)
```

Reply:
We can also consider retaining the same Module name as ReBloom by renaming this to `bf`. 

In case any client uses the `INFO MODULES` or `HELLO` command, they will see the same name.

Reply:
@enjoy-binbin @hpatro - Updated the section.

The Module name is "bf" to be compatible with HELLO, INFO, MODULE LIST operations.

The current Module data type name is just "bloom" (different from ReBloom) because it is not RDB compatible with ReBloom.

Reply:
Can you add this explanation in the RFC?

Reply:
The explanation was added, but the name was updated to `bloomfltr` btw.